File: src/snippets/rate-limiting/reference/openai-agents/ByUserId.ts
import { tokenBucket } from "@arcjet/guard";

const lookupLimit = tokenBucket({
  bucket: "lookups",
  refillRate: 5,
  intervalSeconds: 10,
  maxTokens: 10,
});

const userId = "user123";
// Key the bucket on a trusted application identity, not a model argument.
const rules = [lookupLimit({ key: userId, requested: 5 })];

File: src/snippets/rate-limiting/reference/claude-managed-agents/py/ByUserId.py
from arcjet.guard import TokenBucket

lookup_limit = TokenBucket(
    refill_rate=5,
    interval_seconds=10,
    max_tokens=10,
    bucket="lookups",
)

user_id = "user123"
# Key the bucket on a trusted application identity, not a model argument.
rules = [lookup_limit(key=user_id, requested=5)]
